### ISSUE TITLE
messages not being sent or received consistently

### ISSUE THREAD

**Issue by james-see** [AnemoneLabs/unmessage#23]
I tried this with a colleague and was able to install and use the cli version successfully and connect and chat with him. Unfortunately our messages were not getting through to each other successfully 90% of the time. 

To be more specific, it was stating that it was sending messages but we were not receiving them. Would be nice to have a verbose mode in the cli or a debugging mode to see all the information.

/verify worked right away and it took two times to get a /req-send to work

**Comment by felipedau** [comment]
Hi @jamesacampbell, thanks for trying it!

> I tried this with a colleague and was able to install and use the cli version successfully and connect and chat with him.

Good to hear that. Also, did you just not use the GUI or you had problems with it?

> Unfortunately our messages were not getting through to each other successfully 90% of the time.
> 
> To be more specific, it was stating that it was sending messages but we were not receiving them. Would be nice to have a verbose mode in the cli or a debugging mode to see all the information.
> 
> /verify worked right away and it took two times to get a /req-send to work

Did you get "the other party's Onion Service is not online or has not been established yet" the first time that you used `/req-send`?

That's curious. unMessage sometimes fails to connect to a user's Onion Service, but after the connection is established all messages should be delivered succesfully. What you are saying is that the CLI was displaying to the sender `alice< hello!` but not `bob: hello!` to the receiver?

You are right, unMessage needs both better messages for the user, as well as a logger or something like that. I am going to open issues for that as well.

`/verify` is just a "local" command so you should not have problems with that. If you can, it would be interesting to test `/auth` because that exchanges a few packets back and forth. It usually takes 5~10 seconds to complete after the receiver enters the secret.

Thank you very much for the feedback!

**Comment by felipedau** [comment]
@jamesacampbell, unMessage 0.2.0 was recently released and the biggest change is that it uses Ephemeral Onion Services and will enable the UI only after it finishes bootstrapping and the peer's Onion Service is reachable. I believe that the issue you had was related to this, as the previous version did not have these features. I have not been able to reproduce what you reported.

You can update it with `pip install --upgrade unmessage`, but you will need to `rm -rf ~/.config/unMessage/` because there are some incompatibility between the user file structures. Then you will have to reestablish the conversations as well. Sorry about that :/

Just a reminder: only one peer needs to send the request. The other one will then receive it and reply to complete the process.

Thanks!

**Comment by felipedau** [comment]
Well, I forgot new dependencies were added, so before the `install --upgrade` command, make sure you install the dependencies in the [README](https://github.com/AnemoneLabs/unmessage) and you have `tor` >= 0.2.7.1.

Thanks!

**Comment by james-see** [comment]
@felipedau thanks! There was one other snag. Package `PyAudio` failed to install. I had to install portaudio via `brew install portaudio` on OSX and then did the `install --upgrade` and was fine.

**Comment by felipedau** [comment]
> @felipedau thanks! There was one other snag. Package PyAudio failed to install. I had to install portaudio via brew install portaudio on OSX and then did the install --upgrade and was fine.

Thanks @jamesacampbell! So it successfully bootstrapped  and established a conversation on OSX? I do not think we had tested on OSX before :O

Do you think you could write installation instructions for OSX (similar to the ones on the README) so we could add it to the docs?

Thanks!

**Comment by james-see** [comment]
@felipedau yes, will do on the instructions and will post it here. I also had to install one other dependency via homebrew, `brew install opus`. Then I was able to successfully start unmessage-cli and worked fine. I didn't fully test all of the features, but will try this week. Thanks

**Comment by felipedau** [comment]
@jamesacampbell, that's great!

BTW, unTalk (the voice feature) is pretty new and we have not used it with many audio devices or even on OSX, so I bet that you will have issues with that - at least to find out the right devices to use. If you want to test it, there are some [steps](https://github.com/AnemoneLabs/unmessage/issues/13#issuecomment-280894609) to do that. Also, feel free to ask anything.

Thank you very much :)
